modules/prokka/prokka.nf

```diff
@@ -1,19 +1,44 @@
+nextflow.enable.dsl = 2
 
 process PROKKA {
     tag { pair_id }
     publishDir "${params.output_dir}/prokka", mode: 'copy'
 
     input:
-    tuple val(pair_id), path("${pair_id}.contigs.fa")
+    tuple val(pair_id), path(contigs_file)
+    val(stain_genus_species)
 
     output:
     path("${pair_id}_prokka")
 
     script:
 
     prokka_reference_argument = ""
-    if (params.prokka_reference) {
-        prokka_reference_argument = "--proteins ${params.prokka_reference}"
+
+    sketch_string = stain_genus_species.toString().split("\\t")
+
+    stain = sketch_string[0].strip()
+    genus = sketch_string[1].strip()
+    species = sketch_string[2].strip()
+
+    prokka_gramstain_argument = ""
+
+    if (stain == "pos") {
+        prokka_gramstain_argument = "--gram pos"
+    } else if (stain == "neg") {
+        prokka_gramstain_argument = "--gram neg"
+    } else {
+        prokka_gramstain_argument = ""
+    }
+
+    prokka_genus_argument = ""
+    prokka_species_argument = ""
+
+    if (genus != "Multiple") {
+        prokka_genus_argument = "--genus ${genus}"
```

Comment:
This was the bug which took me a while today, we need to trim the strings and then use those.